src/cobra/test/test_core/test_core_reaction.py

```diff
@@ -39,14 +49,13 @@ def test_gpr() -> None:
         assert reaction.gene_reaction_rule == "(b1 and b2) or (b3 and b4)"
         assert len(reaction.genes) == 4
 
-    # Ensure regular expressions correctly extract genes from malformed
-    # GPR string
+    # Malformed GPR strings will lead to empty GPRs with no genes
     with warnings.catch_warnings():
         warnings.simplefilter("ignore")
         reaction.gene_reaction_rule = "(a1 or a2"
-        assert len(reaction.genes) == 2
+        assert len(reaction.genes) == 0
         reaction.gene_reaction_rule = "(forT or "
-        assert len(reaction.genes) == 1
+        assert len(reaction.genes) == 0
```

Comment:
Couple of thoughts:

1. Should a malformed GPR really only be a warning and not an error?
2. I would make this a separate, parametrized test.
3. Instead of filtering the warning, maybe we should ensure that it exists? If we don't decide that an exception would be better that is. https://docs.pytest.org/en/latest/how-to/capture-warnings.html#warns

Reply:
1. I don't have strong feelings about this. Since the previous version allowed it, and actually extracted the genes, I was leading to warning. But it could be an error, since the behavior changed.
2 & 3. Done.

Reply:
@cdiener what's your experience? I can see that a model not loading from SBML due to a malformed GPR would be extremely annoying. However, when a user sets a GPR, I prefer the error to let them know right away that it's wrong. Maybe the model parser should catch the exception and turn it into a warning?